ReSharper.FSharp/src/FSharp.Psi.Features/src/FSharpTypingAssist.fs

```diff
@@ -1493,6 +1495,47 @@ type FSharpTypingAssist(lifetime, solution, settingsStore, cachingLexerService,
 
         insertCharInBrackets context lexer atChars typedQuotationBrackets LeftBracketOnly.No
 
+    member x.HandleXmlDocTag(context: ITypingContext) =
+        let textControl = context.TextControl
+        let offset = textControl.Caret.Offset()
+        let mutable lexer = Unchecked.defaultof<_>
+        if not (x.GetCachingLexer(textControl, &lexer)) then false else
+
+        if offset < 4 then false else
+
+        // Check if new doc-comment block is being started
+        context.CallNext()
+        let file = x.CommitPsiOnlyAndProceedWithDirtyCaches(textControl, id)
```

Comment:
Is there any need to always commit Psi here? It's happening synchronously on the UI thread, and it would be nice to only do this if we checked the token first and decided that it was likely that we'd need to generate the block.

Reply:
>and it would be nice to only do this if we checked the token first and decided that it was likely that we'd need to generate the block.

Yes, we can add a such check